In [ ]:
##Task 4.6. Combinig and Exporting Data

In [2]:
# Import libraries

import pandas as pd
import numpy as np
import os

In [3]:
path=r'C:\Users\User\Achievement4\Data'

In [4]:
# Import "orders_products_combined" file
df_ords_prods_combined = pd.read_pickle(os.path.join(path, 'Prepared Data', 'orders_products_combined.pkl'))



In [24]:
# Import dataset "products_clean"
df_prods = pd.read_csv(os.path.join(path, 'Prepared Data', 'products_clean.csv'), index_col = False)

In [6]:
#Check the shape of the imported dataframe
df_ords_prods_combined.shape

(32434489, 11)

###The shape is the same as the shape of the exported file

In [28]:
df_ords_prods_combined.head()

order_id  user_id  order_number  orders_day_of_week  hour_of_day  \
0   2539329        1             1                   2            8   
1   2539329        1             1                   2            8   
2   2539329        1             1                   2            8   
3   2539329        1             1                   2            8   
4   2539329        1             1                   2            8   

   days_since_prior_order  first_order  product_id  add_to_cart_order  \
0                     NaN         True         196                  1   
1                     NaN         True       14084                  2   
2                     NaN         True       12427                  3   
3                     NaN         True       26088                  4   
4                     NaN         True       26405                  5   

   reordered  
0          0  
1          0  
2          0  
3          0  
4          0

In [8]:
#Checking the "products_clean" file
df_prods.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [9]:
df_prods.shape

(49672, 6)

In [10]:
#Dropping 'Unnamed' column in "df_prods"
df_prods=df_prods.drop(columns=['Unnamed: 0' ])

In [11]:
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [ ]:
###The both dataframes have 1 common column - "product_id", thus, we can merge them (using an inner join by default).

In [25]:
#Merging both Dataframes
df_ords_prods_merged=df_ords_prods_combined.merge(df_prods, on = 'product_id', indicator=True)

In [15]:
df_ords_prods_merged.head()

order_id  user_id  order_number  orders_day_of_week  hour_of_day  \
0   2539329        1             1                   2            8   
1   2539329        1             1                   2            8   
2   2539329        1             1                   2            8   
3   2539329        1             1                   2            8   
4   2539329        1             1                   2            8   

   days_since_prior_order  first_order  product_id  add_to_cart_order  \
0                     NaN         True         196                  1   
1                     NaN         True       14084                  2   
2                     NaN         True       12427                  3   
3                     NaN         True       26088                  4   
4                     NaN         True       26405                  5   

   reordered                             product_name  aisle_id  \
0          0                                     Soda        77   
1          0  Organic Unsweetened Vanilla Almond Milk        91   
2          0                      Original Beef Jerky        23   
3          0               Aged White Cheddar Popcorn        23   
4          0         XL Pick-A-Size Paper Towel Rolls        54   

   department_id  prices _merge  
0              7     9.0   both  
1             16    12.5   both  
2             19     4.4   both  
3             19     4.7   both  
4             17     1.0   both

In [26]:
#Confirm the results of the merge using the merge flag
df_ords_prods_merged['_merge'].value_counts()

_merge
both          32404859
left_only            0
right_only           0
Name: count, dtype: int64

###The merged dataframe contains 32,404,859 rows, with each row holding data from both input datasets, as we applied an inner join for this merge.

In [23]:
# Export data to pkl
df_ords_prods_merged.to_pickle(os.path.join(path, 'Prepared Data', 'ords_prods_merged.pkl'))